src/main/scala/com/ing/wbaa/gargoyle/proxy/api/ProxyService.scala

```diff
@@ -35,34 +37,26 @@ trait ProxyService extends LazyLogging {
         extractRequest { httpRequest =>
           extracts3Request { s3Request =>
 
-            onComplete(areCredentialsAuthentic(s3Request.credential)) {
-              case Success(true) =>
-                logger.debug(s"Request authenticated: $s3Request")
-
-                onComplete(getUserForAccessKey(s3Request.credential)) {
-                  case Success(Some(userSTS: User)) =>
-                    logger.debug(s"User retrieved: $userSTS")
-
-                    if (isUserAuthorizedForRequest(s3Request, userSTS)) {
-                      logger.info(s"User (${userSTS.userName}) successfully authorized for request: $s3Request")
-                      complete(executeRequest(httpRequest, remoteAddress, userSTS))
-                    } else {
-                      val msg = s"User (${userSTS.userName}) not authorized for request: $s3Request"
-                      logger.warn(msg)
-                      complete((StatusCodes.Unauthorized, msg))
-                    }
-
-                  case Success(None) =>
-                    val msg = s"User not found for accesskey: ${s3Request.credential.accessKey.value}"
-                    logger.warn(msg)
-                    complete((StatusCodes.Unauthorized, msg))
-
-                  case Failure(exception) =>
-                    logger.error("An error occurred retrieving the User from STS service", exception)
-                    complete(StatusCodes.InternalServerError)
+            onComplete(areCredentialsActive(s3Request.credential)) {
+              case Success(Some(userSTS: User)) =>
+                logger.debug(s"Credentials active for request, user retrieved: $userSTS")
+
+                if (isUserAuthenticated(httpRequest, userSTS.secretKey)) {
+                  logger.debug(s"Request authenticated: $httpRequest")
+
+                  if (isUserAuthorizedForRequest(s3Request, userSTS)) {
```

Comment:
I am hitting this with npa user, seq:
Proxy
Signertype used: Some(AWS)
Extracted S3 Request: 
S3Request(AwsRequestCredential(AwsAccessKey(accesskey),None),Some(admin),Some(user),Read)
Credentials active for request, user retrieved:
Request authenticated: HttpRequest
checking ranger with request: RangerAccessRequestImpl=
User (UserName(ceph-admin)) not authorized for request

Sts
isCredentialActive ok for accessKey=accesskey, sessionToken=None

Reply:
Seems like there is no ranger entry that allows `ceph-admin` to do this Read right?

Reply:
Problem related to how plugin forms url to talk to RGW (radosgw-4j library related)